rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -37,23 +45,23 @@
             example = "labels: \n\tlabel-one: \"value-one\"")
     @Language("yml")
     String yaml;
+    @Option(displayName = "Filter YAML snippet",
+            description = "The YAML snippet to to use as a basis for filtering which files are applicable to this recipe.",
+            example = "labels: \n\tlabel-two: \"value-two\"",
+            required = false)
+    @Language("yml")
+    @Nullable
+    String filterYaml;
```

Comment:
Likely this `filterYaml` is an applicable test we would use in many other places. Extract it?

Reply:
I've already found another place in rewrite-kubernetes where I need this, so definitely need to make it reusable.